src/routes/game/components/dialogs/components/gameOver/GameOverDialog.vue

```diff
@@ -0,0 +1,346 @@
+      rightPlayerRematch,
+      rightPlayerWins,
+      leftPlayerWonMatch,
+      rightPlayerWonMatch,
+    };
+  },
+  data() {
+    return {
+      leavingGame: false,
+      showSnackbar: false,
+      snackBarMessage: '',
+      snackColor: 'error',
+    };
+  },
+  computed: {
+    show: {
+      get() {
+        return this.modelValue;
+      },
+      set() {
+        // do nothing - parent controls whether dialog is open
+      },
+    },
+    ...mapStores(useGameStore),
+    heading() {
+      if (this.gameStore.isSpectating) {
+        return this.spectatorHeading;
+      }
+      if (this.matchIsOver) {
+        // You win the match / you lose the match
+        const messageName = this.leftPlayerWonMatch ? 'game.dialogs.gameOverDialog.youWinTheMatch' : 'game.dialogs.gameOverDialog.youLoseTheMatch';
+        return this.t(messageName);
+      }
+      // Draw / You Won / You Lose
+      const messageName = this.stalemate ? 'game.dialogs.gameOverDialog.draw' : this.playerWinsGame ? 'game.dialogs.gameOverDialog.youWin' : 'game.dialogs.gameOverDialog.youLose';
+      return this.t(messageName);
+    },
+    spectatorHeading() {
+      const prefix = 'game.dialogs.gameOverDialog';
+      if (this.gameStore.winnerPNum === null) {
+        return this.t(`${prefix}.draw`);
+      }
+      const winner = this.gameStore.players[this.gameStore.winnerPNum];
+      const res = winner.id === this.gameStore.currentMatch?.games[0].p0 ?
+        this.t(`${prefix}.p1Wins`) : this.t(`${prefix}.p2Wins`);
+      const matchTranslation = this.t(`${prefix}.match`);
+      return this.matchIsOver ? `${res} ${matchTranslation}` : res;
+    },
+    headingDataAttr() {
+      if (this.stalemate) {
+        return 'stalemate-heading';
+      }
+
+      if (this.gameStore.isSpectating) {
+        const origianlP0 = this.gameStore.currentMatch?.games[0].p0;
+        const winner = this.gameStore.players[this.gameStore.winnerPNum];
+        return winner.id === origianlP0 ? 'p1-wins-heading' : 'p2-wins-heading';
+      }
+
+      return this.playerWinsGame ? 'victory-heading' : 'loss-heading';
+    },
+    latestResult() {
+      if (this.stalemate) {
+        return 'Stalemate';
+      }
+      if (this.gameStore.isSpectating) {
+        const origianlP0 = this.gameStore.currentMatch?.games[0].p0;
+        const winner = this.gameStore.players[this.gameStore.winnerPNum];
+        return winner.id === origianlP0 ? 'Won' : 'Lost';
+      }
+
+      return this.playerWinsGame ? 'Won' : 'Lost';
+    },
+    isMobilePortrait() {
+      return this.$vuetify.display.xs;
+    },
+    currentMatch() {
+      return this.gameStore.currentMatch;
+    },
+    matchGameStats() {
+      const currentMatchGames = this.gameStore.currentMatch?.games ?? [];
+      return currentMatchGames.map((game) => {
+        if (game.status === GameStatus.FINISHED){
+          return game.winner === null ? 'D' : game.winner === this.gameStore.opponent.id ? 'L' : 'W';
```

Comment:
I don't really like nested ternaries, I would switch this up.

```
return currentMatchGames.map((game) => {
  if (game.status !== GameStatus.FINISHED){
    return 'I';
  }
  if (game.winner === null) {
    return 'D';
  }

  if (game.winner === this.gameStore.opponent.id) {
    return 'L'
  }
  
  return 'W';
});
```

If you do this in multiple places convert it to a utility or something and reuse it.

Reply:
1. I think you're right that it reads better if we remove the nested ternary. 
2. I think a switch statement would be nice since we have multiple comparisons to the same value
3. Most importantly, looking again, this is unused lol. I'm ditching it.

But if we _did_ want this, I'd think style wise this would be the sweet spot:

```
return currentMatchGames.map((game) => {
  if (game.status !== GameStatus.FINISHED){
    return 'I';
  }

  switch(game.winner) {
    case this.gameStore.player.id:
      return 'W';
    case this.gameStore.opponent.id:
      return 'L';
    default:
      return 'D';
  }
});
```